static/src/javascripts/projects/common/modules/identity/api.ts

```diff
@@ -0,0 +1,431 @@
+import { storage } from '@guardian/libs';
+import { mergeCalls } from 'common/modules/async-call-merger';
+import {
+	createAuthenticationComponentEvent,
+	createAuthenticationComponentEventParams,
+} from 'common/modules/identity/auth-component-event-params';
+import config_ from 'lib/config';
+import { getCookie as getCookieByName } from 'lib/cookies';
+import fetchJson from 'lib/fetch-json';
+import mediator from 'lib/mediator';
+import { getUrlVars } from 'lib/url';
+
+// This is really a hacky workaround ⚠️
+const config = config_ as {
+	get: (s: string) => string;
+};
+
+// Types info coming from https://github.com/guardian/discussion-rendering/blob/fc14c26db73bfec8a04ff7a503ed9f90f1a1a8ad/src/types.ts
+
+type PasswordCredential = {
+	id: string;
+	password: string;
+};
+
+type SettableConsent = {
+	id: string;
+	consented: boolean;
+};
+
+type Newsletter = {
+	id: string;
+	subscribed?: boolean;
+};
+
+type UserNameError = {
+	message: string;
+	description: string;
+	context: string;
+};
+
+type UserConsents = {
+	id: string;
+	actor: string;
+	version: number;
+	consented: boolean;
+	timestamp: string;
+	privacyPolicyVersion: number;
+};
+
+type UserGroups = {
+	path: string;
+	packageCode: string;
+};
+
+export type IdentityUser = {
+	dates: { accountCreatedDate: string };
+	consents: UserConsents[];
+	userGroups: UserGroups[];
+	publicFields: {
+		username: string;
+		displayName: string;
+	};
+	statusFields: {
+		userEmailValidated: boolean;
```

Comment:
I have a `hasRepermissioned` attribute with a `boolean` type value in `statusFields`

Reply:
Added in d010572ce4